core/kazoo_directory/src/kz_directory_endpoint.erl

```diff
@@ -86,36 +86,20 @@ endpoint_id(Endpoint, CCVs) ->
                               )
     end.
 
-
--spec maybe_fix_presence_id_realm(kz_term:api_ne_binary(), kz_json:object(), kz_json:object()) -> kz_json:object().
-maybe_fix_presence_id_realm('undefined', _Endpoint, CCVs) -> CCVs;
-maybe_fix_presence_id_realm(PresenceId, Endpoint, CCVs) ->
-    case binary:match(PresenceId, <<"@">>) of
-        'nomatch' ->
-            Realm = get_realm(Endpoint),
-            kz_json:set_value(<<"Presence-ID">>, <<PresenceId/binary, $@, Realm/binary>>, CCVs);
-        _Else -> kz_json:set_value(<<"Presence-ID">>, PresenceId, CCVs)
-    end.
-
--spec presence_id(kz_json:object(), kz_json:object()) -> kz_json:object().
-presence_id(Endpoint, CCVs) ->
-    Default = kzd_devices:sip_username(Endpoint),
-    PresenceId = kzd_devices:presence_id(Endpoint, Default),
-    case kz_term:is_empty(PresenceId) of
-        'true' -> maybe_fix_presence_id_realm(Default, Endpoint, CCVs);
-        'false' -> maybe_fix_presence_id_realm(PresenceId, Endpoint, CCVs)
-    end.
-
 -spec owner_id(kz_json:object(), kz_json:object()) -> kz_json:object().
 owner_id(Endpoint, CCVs) ->
-    case kz_json:get_ne_binary_value(<<"owner_id">>, Endpoint) of
+    case kzd_devices:owner_id(Endpoint) of
         'undefined' -> CCVs;
         OwnerId -> kz_json:set_value(<<"Owner-ID">>, OwnerId, CCVs)
     end.
 
+-spec presence_id(kz_json:object(), kz_json:object()) -> kz_json:object().
+presence_id(Endpoint, CCVs) ->
+    kz_json:set_value(<<"Presence-ID">>, kzd_devices:calculate_presence_id(Endpoint), CCVs).
```

Comment:
will revisit this and probably remove as the endpoint will already have merged the hotdesked user